compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -1672,7 +1672,17 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         let mut seen_fields = FxHashMap::default();
 
-        let mut error_happened = false;
+        let mut error_happened = if variant.fields.len() > remaining_fields.len() {
+            // Some field is defined more than once. Make sure we don't try to
+            // instantiate this struct in static/const context.
+            let guar =
+                self.dcx().span_delayed_bug(expr.span, "struct fields have non-unique names");
+            self.set_tainted_by_errors(guar);
+
+            true
+        } else {
+            false
+        };
```

Comment:
Can you split this up like so:

```suggestion
        let mut error_happened = false; 
        
        if variant.fields.len() != remaining_fields.len() {
            // Some field is defined more than once. Make sure we don't try to
            // instantiate this struct in static/const context.
            let guar =
                self.dcx().span_delayed_bug(expr.span, "struct fields have non-unique names");
            self.set_tainted_by_errors(guar);
            error_happened = true;
        }
```